src/main/java/com/conveyal/gtfs/loader/JdbcTableWriter.java

```diff
@@ -1452,6 +1463,49 @@ private static String getValueForId(int id, String fieldName, String namespace,
         return value;
     }
 
+    /**
+     * Parse a list field in the schedule exceptions table (dates, added_service, or removed_service) into an array of strings.
+     * List fields take the form {ServiceId1, ServiceId2, ServiceId3, ...}
+     */
+    private String[] parseExceptionListField(int id, String namespace, Table table, String type) throws SQLException {
+        String parsedString;
+        if (type.equals("service")) {
+            String addedServiceIds = getValueForId(id, "added_service", namespace, table, connection);
+            String removedServiceIds = getValueForId(id, "removed_service", namespace, table, connection);
+            parsedString = addedServiceIds + "," + removedServiceIds;
+        } else {
+            parsedString = getValueForId(id, type, namespace, table, connection);
+        }
+        return parsedString.replaceAll("[{}]", "").split("[,]", 0);
+    }
+
+    /**
+     * Delete all entries in calendar dates associated with a schedule exception.
+     */
+    private Integer deleteCalendarDatesForException(
+        int id,
+        String namespace,
+        Table table,
+        String refTableName
+    ) throws SQLException {
+        String[] serviceIds = parseExceptionListField(id, namespace, table, "service");
+        String[] dates = parseExceptionListField(id, namespace, table, "dates");
+        int resultCount = 0;
+
+        // Schedule exceptions map to many different calendar dates (one for each date / service ID combination)
+        for (String date : dates) {
+            for (String serviceId : serviceIds) {
+                String sql =String.format("delete from %s where service_id = ? and date = ?", refTableName);
+                PreparedStatement updateStatement = connection.prepareStatement(sql);
```

Comment:
The point of prepared statements is that you can call them multiple time with different params, so move this declaration outside of the `for` loops.

Reply:
Updated in https://github.com/conveyal/gtfs-lib/pull/376/commits/577ae945553f3c5f6f2a480ae4a0588b5c40e116